test/library/metrics/query.go

```diff
@@ -0,0 +1,107 @@
+package metrics
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/prometheus/common/model"
+)
+
+type PrometheusClient struct {
+	// Host address of Prometheus public route
+	host string
+	// ServiceAccount bearer token to pass though OpenShift oauth proxy
+	token string
+}
+
+func NewPrometheusClient(client *kubernetes.Clientset, rc *routeclient.Clientset) (*PrometheusClient, error) {
+	_, err := client.CoreV1().Services("openshift-monitoring").Get("prometheus-k8s", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	route, err := rc.RouteV1().Routes("openshift-monitoring").Get("prometheus-k8s", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	host := route.Spec.Host
```

Comment:
this is bad, use `route.Status` to retrieve the host from `RouteIngress`

Reply:
done, w/ route.Status.Ingress[0].Host